pybedlite/overlap_detector.py

```diff
@@ -56,6 +80,35 @@
 from pybedlite.bed_source import BedSource
 
 
+class GenomicSpan(Hashable, Protocol):
+    """
+    A structural type for a span on a reference sequence with zero-based open-ended coordinates.
+    """
+
+    @property
+    def refname(self) -> str:
+        """A reference sequence name."""
+
+    @property
+    def start(self) -> int:
+        """A 0-based start position."""
+
+    @property
+    def end(self) -> int:
+        """A 0-based open-ended position."""
+
+
+class StrandedGenomicSpan(GenomicSpan, Protocol):
+    """
+    A structural type for a stranded span on a reference sequence with zero-based open-ended
+    coordinates.
+    """
+
+    @property
+    def negative(self) -> bool:
+        """True if the interval is on the negative strand, False otherwise"""
```

Comment:
We considered this and could still go in this direction.

The benefit of having `negative` on `BedRecord` is it requires no changes to `Interval` and one additional helper method on `BedRecord`. But if we want to use `is_negative` then we would add that as a helper to `BedRecord` and would also need to add a **_duplicate getter_** to `Interval` (it already has a `negative(): bool` field) making more API bloat. I suppose we could break the API fully and rename the fields, but I don't have a good grasp on how much downstream/user code would have to change and it seems like we're trying to avoid that as completely as possible.

Reply:
My 2c would be to change the API now (we're on v0.1.0, I don't think there are high expectations for API stability 🙂 )

I'd be in favor of 
1. renaming `Interval`'s `negative` attribute to `is_negative`, 
2. adding a `negative(): bool` property which raises a `DeprecationWarning` and returns the value of `is_negative`, for backwards compatibility
3. cycling the `negative` property out in a few releases.

Reply:
If we do this, then I would also bundle breaking the API for `refname`/`chrom` as the primary property for the reference sequence too. I recommend we defer this to a future PR, though, and break everything all at once with the deprecation warnings as you recommend.